### ISSUE TITLE
数据泄露风险

### ISSUE THREAD

**Issue by OstrichThinking** [thuml/Time-Series-Library#816]
在进行分类模型训练时：

def train(self, setting):
        train_data, train_loader = self._get_data(flag='TRAIN')
        vali_data, vali_loader = self._get_data(flag='TEST')
        test_data, test_loader = self._get_data(flag='TEST')

def test(self, setting, test=0):
        test_data, test_loader = self._get_data(flag='TEST')
        if test:

为训练集、测试集分别初始化 Dataloader， 每次初始化涉及到标准化操作：

# pre_process
normalizer = Normalizer()
self.feature_df = normalizer.normalize(self.feature_df)
print(len(self.all_IDs))

是否存在“训练集用自己均值方差进行标准化、测试集也用自己的均值方差进行标准化”的问题？

**Comment by appleweiping** [comment]
I checked the current main branch at 4e938a1. UEAloader creates a fresh Normalizer for every split, while Exp_Classification constructs TRAIN and TEST datasets separately. TEST is therefore standardized with its own mean and standard deviation.

I can work on this. My preference is to let data_provider and UEAloader accept fitted training statistics and pass them into TEST loaders, instead of loading the TRAIN file again inside every TEST dataset. Would that direction fit the intended API?
